presto-main/src/main/java/com/facebook/presto/metadata/SqlAggregationFunction.java

```diff
@@ -91,39 +102,27 @@ public boolean isDeterministic()
 
     public abstract InternalAggregationFunction specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry);
 
-    public static class SimpleSqlAggregationFunction
+    public static class AnnotationBasedSqlAggregationFunction
             extends SqlAggregationFunction
     {
-        private final InternalAggregationFunction function;
-        private final String description;
+        private final BindableAggregationFunction aggregationSignature;
```

Comment:
As in later comment - BindableAggregation function is top level.
In the future, it requires refactor to only have one responsibility (of holding aggregation metadata).